crates/libs/metadata/src/reader/flags.rs

```diff
@@ -1,23 +1,54 @@
-#[derive(Default)]
-pub struct ParamFlags(pub u32);
+pub struct FieldAttributes(pub usize);
+pub struct MethodAttributes(pub usize);
+pub struct MethodImplAttributes(pub usize);
+pub struct ParamAttributes(pub usize);
+pub struct PInvokeAttributes(pub usize);
+pub struct TypeAttributes(pub usize);
 
-impl ParamFlags {
+impl FieldAttributes {
+    pub fn literal(&self) -> bool {
+        self.0 & 0x40 != 0
+    }
+}
+
+impl MethodAttributes {
+    pub fn special(&self) -> bool {
+        self.0 & 0x800 != 0
+    }
+}
+
+impl MethodImplAttributes {
+    pub fn preserve_sig(&self) -> bool {
+        self.0 & 0x80 != 0
+    }
+}
+
+impl ParamAttributes {
     pub fn input(&self) -> bool {
-        self.0 & 0x0001 != 0
+        self.0 & 0x1 != 0
     }
     pub fn output(&self) -> bool {
-        self.0 & 0x0002 != 0
+        self.0 & 0x2 != 0
     }
     pub fn optional(&self) -> bool {
-        self.0 & 0x0010 != 0
+        self.0 & 0x10 != 0
     }
 }
 
-#[derive(Default)]
-pub struct PInvokeAttributes(pub u32);
-
 impl PInvokeAttributes {
     pub fn last_error(&self) -> bool {
-        self.0 & 0x0040 != 0
+        self.0 & 0x40 != 0
+    }
+}
+
+impl TypeAttributes {
+    pub fn union(&self) -> bool {
```

Comment:
Nit: This is really `ExplicitLayout` right, which could or could not be related to unions.

Reply:
Yep, I can use the more explicit :wink: name.